contents/swift/welcome-to-swift.md

```diff
@@ -144,9 +146,36 @@ Here are some of the benefits of using `guard` statement over `if-let` statement
 1. Unlike the `if-let` statement, using the `guard` statement causes `past` to remain defined and can be used till the function exits.
 1. While using `if-let` statements can lead to deeply nested `if-let` statements (i.e. pyramid of doom), `guard` statements allow us to have the happy path to be not indented, thereby increasing code readability.
 
-## Data Types
+### Benefit: Handle Post-operations in Advance
+
+The `defer` key word in Swift provides an easy and safe way to execute some code before leaving current scope. Therefore, if you have to do some operations at the end of function, you can now put it at the beginning.
+
+The following code is an example of using a file:
+```swift
+let fileDescriptor = open(url.path, O_EVTONLY)
+
+// Use file descriptor
+
+close(fileDescriptor)
+```
+
+However, with very long code of using file descriptor, you may forget to close the file at the end. Then, `defer` will solve this problem easily:
+
+```swift
+let fileDescriptor = open(url.path, O_EVTONLY)
+
+defer {
+    close(fileDescriptor)
+}
+
+// Use file descriptor
+```
+
+Through this implementation, the file will be closed before we leave this function, and it is unlikely to forget closing file using `defer`.
```

Comment:
Just a small suggestion: I think your example for using `defer` could be even better if you show that `defer` can be used to execute the same code when the function exits from multiple different return points. e.g.

Before `defer:
```swift
let fileDescriptor = open(url.path, O_EVTONLY)

if fileDescriptor == -1 {
	close(fileDescriptor)
	return "Failed"
}

// Do something with file
close(fileDescriptor)
return "Success"
```

After `defer`
```swift
let fileDescriptor = open(url.path, O_EVTONLY)
defer { close(fileDescriptor) }

if fileDescriptor == -1 {
	return "Failed"
}

// Do something with file
return "Success"
```

Just my suggestion, but your explanation is still understandable :)

Reply:
Cool! This is really a better example